src/goto-programs/remove_function_pointers.cpp

```diff
@@ -420,95 +420,26 @@ void remove_function_pointerst::remove_function_pointer(
   const irep_idt &function_id,
   goto_programt::targett target)
 {
-  const code_function_callt &code = target->get_function_call();
-
-  const auto &function = to_dereference_expr(code.function());
+  goto_programt::const_targett const_target = target;
+  const auto functions =
+    get_function_pointer_targets(goto_program, const_target);
 
-  // this better have the right type
-  code_typet call_type=to_code_type(function.type());
-
-  // refine the type in case the forward declaration was incomplete
-  if(call_type.has_ellipsis() &&
```

Comment:
I'm not quite sure that this behaviour has been carried over with the changes.  Is this handled by one of the new branches?

Reply:
Yes, it was moved to a dedicated method `refine_call_type` which is now called during init-state of `get_function_pointer_targets`.